docker-compose.yaml

```diff
@@ -31,124 +42,158 @@ services:
     depends_on:
       - server-connect
       - server-grpc
-  client-connect-to-server-connect-h1:
+  client-connect-go-to-server-connect-go-h1:
     build:
       context: .
-      dockerfile: Dockerfile.crosstest
+      dockerfile: Dockerfile.crosstestgo
       args:
         TEST_CONNECT_GO_BRANCH: "${TEST_CONNECT_GO_BRANCH:-}"
     entrypoint: /usr/local/bin/client --host="server-connect" --port="8080" --implementation="connect-h1" --cert "cert/client.crt" --key "cert/client.key"
     depends_on:
       - server-connect
-  client-connect-to-server-connect-h2:
+  client-connect-go-to-server-connect-go-h2:
     build:
       context: .
-      dockerfile: Dockerfile.crosstest
+      dockerfile: Dockerfile.crosstestgo
       args:
         TEST_CONNECT_GO_BRANCH: "${TEST_CONNECT_GO_BRANCH:-}"
     entrypoint: /usr/local/bin/client --host="server-connect" --port="8081" --implementation="connect-h2" --cert "cert/client.crt" --key "cert/client.key"
     depends_on:
       - server-connect
-  client-connect-to-server-connect-h3:
+  client-connect-go-to-server-connect-go-h3:
     build:
       context: .
-      dockerfile: Dockerfile.crosstest
+      dockerfile: Dockerfile.crosstestgo
       args:
         TEST_CONNECT_GO_BRANCH: "${TEST_CONNECT_GO_BRANCH:-}"
     entrypoint: /usr/local/bin/client --host="server-connect" --port="8082" --implementation="connect-h3" --cert "cert/client.crt" --key "cert/client.key"
     depends_on:
       - server-connect
-  client-connect-grpc-to-server-connect-h1:
+  client-connect-go-to-server-connect-node-fastify-h1:
+    build:
+      context: .
+      dockerfile: Dockerfile.crosstestgo
+      args:
+        TEST_CONNECT_GO_BRANCH: "${TEST_CONNECT_GO_BRANCH:-}"
+    command: >
+        /bin/sh -c "
+        echo Waiting for Fastify to start...;
+        while ! nc -z server-connect-node-fastify 8084;
+        do
+          sleep 1;
+        done;
+        /usr/local/bin/client --host='server-connect-node-fastify' --port='8084' --implementation='connect-h1' --insecure
+        "
```

Comment:
Super hacky, but the tests seem to fail unless we add some sort of wait to the fastify container. The `depends-on` attribute waits until the container is up, but it seems it takes a bit longer for Fastify to bind to the port. Without something like this, we get a `Connection refused` on the container. Open to suggestions for how to fix this better.

Reply:
What about [healthcheck](https://docs.docker.com/compose/compose-file/05-services/#healthcheck)?

Reply:
I looked at that actually, but I need to test multiple ports (`8084` and `8085`) and there's no way to test multiple ports in docker-compose (or in the Dockerfile). I could create two containers, one for each port but that seemed like overkill.

Reply:
You don't need to listen for multiple ports if they are bound in sequence (they are). Here is an example: https://github.com/bufbuild/connect-crosstest/pull/479/files

The important part is that the health check is defined on the service we want to check, not on multiple clients. This is a definite improvement in my book.

I actually think it would be preferable to treat each run of the test suite as a discrete unit. That also means listening to only one port from a process. Out of scope for this PR IMO.